zkevm-circuits/src/mpt_circuit/helpers.rs

```diff
@@ -261,7 +261,7 @@ pub(crate) fn ext_key_rlc_expr<F: Field>(
                 }).collect::<Vec<_>>());
                 calc_rlc(cb, &key_bytes, 1.expr())
             },
-            and::expr(&[is_long.expr(), is_key_odd.expr()]) => {
+            and::expr(&[is_long.expr(), or::expr(&[and::expr(&[not!(is_key_odd.expr()), not!(is_key_part_odd.expr())]), and::expr(&[is_key_odd.expr(), is_key_part_odd.expr()])])]) => {
```

Comment:
```suggestion
            and::expr(&[is_long.expr(), not!(xor::expr(is_key_odd.expr(), is_key_part_odd.expr()))]) => {
```

Also helps to make it more clear it's the `else` clause. 

Hmmm maybe it's even better if it's split up in `is_long` and `is_short` cases, and then the `is_long` case contains an `if xor::expr(is_key_odd.expr(), is_key_part_odd.expr()) { } else { }`.

Reply:
Changed in https://github.com/privacy-scaling-explorations/zkevm-circuits/pull/1685/commits/4acc57394e92f71d24b4622c78d885a01a698a3d